### ISSUE TITLE
"Featured Image" box not working

### ISSUE THREAD

**Issue by viktormorales** [johnbillion/network-media-library#29]
The "featured image" meta box is empty in the sites other than the main. If I change the site ID with the add_filter, the new main site will show the "featured image", but the other sites will show an empty box.

**Comment by johnbillion** [comment]
Thanks for the report. Can you send a screenshot and let me know a bit about how your Multisite installation is set up? Are there any PHP notices or warnings showing up in your error log?

**Comment by viktormorales** [comment]
Hi John,

I've created a subfolder network with only 2 sites: a spanish and an english blog.

This is the meta box on the spanish version with no PHP nor JS errors.
![image](https://user-images.githubusercontent.com/1565556/44681616-793fe380-aa17-11e8-82ca-ee518f8a9f9f.png)

**Comment by UnionDesign** [comment]
I'm seeing the same issue. Here are the steps to recreate the issue:

1. Add "image.png" to the "parent" media library (in my case the main blog not ID 2)
2. Add post to "blog b" (any child blog) and set it's featured image to "image.png"
3. Switch back to the main blog and delete "image.png" from the media library
4. Switch back to "blog b" and the feature image is gone (as expected) but so is the link to set the featured image. The link is actually in the source code but the <a></a> is empty so there's nothing to click.

So basically when you delete an image from the main media library that's being used / referenced as a featured image somewhere else (i.e. in another blog) the "Set Featured Image" link breaks.

**Comment by UnionDesign** [comment]
p.s. admin_post_thumbnail_html() seems to be the culprit

**Comment by johnbillion** [comment]
Thanks for the details. I'll try to find some time to look into this.

**Comment by UnionDesign** [comment]
Ok I have a solution. Not sure how solid the code is but it's working for me. When you delete an image from the main library that's used as a featured image on a subsite the post doesn't know that the image has been deleted. So has_post_thumbnail still returns as true. So I added a tiny bit of code in your admin_post_thumbnail_html() function to check for the existence of a feature image URL at around line 204

`	if ( $has_thumbnail_url == false ) {
		$search  = 'class="thickbox"></a>';
		$replace = 'class="thickbox">' . esc_html( $post_type_object->labels->set_featured_image ) . '</a>';
	
	} else {
		$search  = '<p class="hide-if-no-js"><a href="#" id="remove-post-thumbnail">SEARCH</a></p>';
		$replace = '<p class="hide-if-no-js"><a href="#" id="remove-post-thumbnail">REPLACE' . esc_html( $post_type_object->labels->remove_featured_image ) . '</a></p>';
	}
	
	$content = str_replace( $search, $replace, $content );
`
